lib/lambda_ethereum_consensus/fork_choice/helpers.ex

```diff
@@ -265,6 +268,47 @@ defmodule LambdaEthereumConsensus.ForkChoice.Helpers do
     end
   end
 
+  @spec finality_checkpoins_by_id(state_id()) ::
+          {:ok, finality_info()} | {:error, String.t()} | :not_found | :empty_slot | :invalid_id
+  def finality_checkpoins_by_id(hex_root) when is_binary(hex_root) do
+    # TODO compute is_optimistic_or_invalid() and is_finalized()
+    execution_optimistic = true
+    finalized = false
+
+    case StateStore.get_state_by_state_root(hex_root) do
+      {:ok, state} ->
+        %{
+          previous_justified_checkpoint: previous_justified_checkpoint,
+          current_justified_checkpoint: current_justified_checkpoint,
+          finalized_checkpoint: finalized_checkpoint
+        } = state
```

Comment:
Maybe do this in the match directly? That is:

```elixir

{:ok,  %{
          previous_justified_checkpoint: previous_justified_checkpoint,
          current_justified_checkpoint: current_justified_checkpoint,
          finalized_checkpoint: finalized_checkpoint
        }} -> 

```

Either that or using the state accessors for the return value, which might be the shortest version:

```elixir
{:ok, state} -> {:ok, 
  {state.previous_justified_checkpoint, 
  state.current_justified_checkpoint, 
  state.finalized_checkpoint, 
  execution_optimistic, 
  finalized}}
```